### PR TITLE
EPMRPP-50472. Logs with attachments.

### PR DESCRIPTION
**Pull request by ElenaRomanchuk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,13 +111,33 @@ cypress.run().then(
...
+
+require('agent-js-cypress/lib/commands/reportPortalCommands');
+
+```
+
+Register ReportPortal plugin (cypress/plugins/index.js):
+
+```javascript
+
+const registerReportPortalPlugin = require('./../../lib/plugin');
```
Comment (AmsterGet) [inline README.md:128]: Please change it to `agent-js-cypress/lib/plugin`

File: example/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// ***********************************************
```
Comment (AmsterGet) [inline example/support/commands.js:1]: Can comments block (`/* */`) be used here?

File: example/integration/example.spec.js [hunk 3]
```diff
@@ -25,6 +25,30 @@ context('Example Cypress Test', () => {
       });
       it('skipped test');
     });
+    context('logs example: Check content Cypress example page', () => {
+      it('report logs', () => {
+        cy.log('cypress log message')
```
Comment (AmsterGet) [inline example/integration/example.spec.js:30]: `;`

File: README.md [hunk 4]
```diff
@@ -124,6 +147,37 @@ Runs support following options:
...
+## ReportPortal custom commands
+
+### Logging
+
+ReportPortal provides the following custom commands for reporting logs into the current test/step.
+
+* log(*message*). Overrides standart Cypress `cy.log(log)`. Reports *message* as an info log of the current test.<br/>
+
+You can use the following methods to report logs and attachments with different log levels:
+* cy.trace (*message* , *file*). Reports *message* and optional *file* as a log of the current test/suite with trace log level.
```
Comment (AmsterGet) [inline README.md:159]: May be better to add `rp` prefix for rp-related methods? What do you think?

File: lib/ipcServer.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+// ***********************************************************
+// This example plugins/index.js can be used to load plugins
+//
+// You can change the location of this file or turn off loading
+// the plugins file with the 'pluginsFile' configuration option.
```
Comment (AmsterGet) [inline lib/ipcServer.js:5]: Is it relevant here?

File: lib/ipcServer.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/plugins-guide
+// ***********************************************************
+
+// This function is called when a project is opened or re-opened (e.g. due to
+// the project's config changing)
+const ipc = require('node-ipc');
+
+const startIPCServer = (subscribeServerEvents) => {
```
Comment (AmsterGet) [inline lib/ipcServer.js:15]: Could you please document (describe in the code with comments) why we needs such additional workaround?

File: README.md (hunk 1 above)
Comment (ElenaRomanchuk) [reply README.md]: Fixed. Thanks!

File: example/support/commands.js (hunk 2 above)
Comment (ElenaRomanchuk) [reply example/support/commands.js]: Fixed

File: README.md (hunk 4 above)
Comment (ElenaRomanchuk) [reply README.md:159]: Decided to leave as is because of overrided cy.log() method.

File: lib/ipcServer.js (hunk 5 above)
Comment (ElenaRomanchuk) [reply lib/ipcServer.js]: Removed

File: lib/ipcServer.js (hunk 6 above)
Comment (ElenaRomanchuk) [reply lib/ipcServer.js:8]: I add the same description in `ipcServer.js` and `ipcClient.js`

**Pull request merged**